hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/protocol/ScmBlockLocationProtocolServerSideTranslatorPB.java

```diff
@@ -94,9 +95,26 @@ public ScmBlockLocationProtocolServerSideTranslatorPB(
         .setTraceID(traceID);
   }
 
+  private boolean isLeader() throws ServiceException {
+    if (!(impl instanceof SCMBlockProtocolServer)) {
+      throw new ServiceException("Should be SCMBlockProtocolServer");
+    } else {
+      return ((SCMBlockProtocolServer) impl).getScm().checkLeader();
+    }
+  }
+
   @Override
   public SCMBlockLocationResponse send(RpcController controller,
       SCMBlockLocationRequest request) throws ServiceException {
+    if (!isLeader()) {
+      SCMBlockLocationResponse.Builder response = createSCMBlockResponse(
+          request.getCmdType(),
+          request.getTraceID());
+      response.setSuccess(false);
+      response.setStatus(Status.SCM_NOT_LEADER);
+      response.setLeaderSCMNodeId(null);
```

Comment:
Should we send back the leader known by this SCM follower instead of null to avoid unnecessary client retry?

Reply:
@xiaoyuyao We decide to implement suggestedLeader in following JIRA https://issues.apache.org/jira/browse/HDDS-4281. To use suggestedLeader from Ratis, we may need a more well-rounded strategy.